celltype_annotation/analysis/cell-assign-sarcoma.Rmd

```diff
@@ -0,0 +1,451 @@
+  sample_id: "SCPCS000262"
+    
+---
+
+This notebook evaluates using `CellAssign` for cell type assignment. 
+Here we are looking specifically at a rhabdomyosarcoma sample that we have known cell type annotations for and comparing those known annotations to using `CellAssign` with two different marker lists. 
+
+- [CellMarker2.0](http://117.50.127.228/CellMarker/)
+- [PanglaoDB](https://panglaodb.se/index.html)
+
+The marker gene lists for both were downloaded and are found in the `references` directory.
+After reading in the marker gene lists, I filtered them to only contain relevant tissue types related to sarcoma (e.g., mucscle, connective tissue).
+The filtered list of cell type markers is turned into a binary gene by cell type matrix with an additional "other" column where no marker genes are found. 
+This is used as input to run `CellAssign` using `scripts/cell-assign.py`. 
+The output of that script is a prediction matrix where each row is a cell barcode and each column is a cell type found in the reference matrix. 
+The corresponding cell type with the top score for each barcode is then assigned to that cell. 
+
+## Set Up
+
+```{r}
+library(SingleCellExperiment)
+library(ggplot2)
+```
+
+```{r}
+# Set up paths 
+
+# build path to annotated sce file 
+processed_data_dir <-file.path("..", "data") 
+processed_sce_file <- glue::glue("{params$library_id}_processed_celltype.rds")
+local_processed_sce_path <- file.path(processed_data_dir, params$sample_id, processed_sce_file)
+
+# anndata output
+anndata_dir <- file.path("..", "data", "anndata")
+anndata_file <- glue::glue("{params$library_id}_anndata.hdf5")
+anndata_file <- file.path(anndata_dir, anndata_file)
+fs::dir_create(anndata_dir)
+
+# reference files for cell marker
+ref_dir <- file.path("..", "references")
+excel_celltype_file <- file.path(ref_dir, "Cell_marker_Human.xlsx")
+panglao_file <- file.path(ref_dir, "PanglaoDB_markers_27_Mar_2020.tsv")
+
+# output matrix files 
+cell_marker_mtx_file <- file.path(ref_dir, "cell_marker_muscle_mtx.csv")
+panglao_matrix_file <- file.path(ref_dir, "panglao_connective_tissue_mtx.csv")
+
+# cellassign predictions
+cellmarker_prediction_file <- file.path(
+  anndata_dir, 
+  glue::glue("{params$library_id}_muscle.tsv")
+)
+panglao_prediction_file <- file.path(
+  anndata_dir, 
+  glue::glue("{params$library_id}_panglao_connective_tissue.tsv")
+)
+
+```
+
+
+```{r}
+# read in all annotated sce 
+processed_sce <- readr::read_rds(local_processed_sce_path)
+```
+
+```{r}
+# write out anndata 
+scpcaTools::sce_to_anndata(processed_sce, anndata_file = anndata_file)
+```
+
+```{r}
+# source helper functions
+function_file <- file.path("..", "utils", "cellassign-helper-functions.R")
+source(function_file)
+```
+
+## CellMarker2.0
+
+```{r}
+# read in excel file
+celltype_markers_df <- openxlsx::read.xlsx(excel_celltype_file)
+```
+
+
+```{r}
+# look at different tissue types available 
+celltype_markers_df |>
+  dplyr::count(tissue_class)
+```
```

Comment:
I'm a bit confused about the organization/origins of this data, so I think a little explanatory paragraph at the start would be useful. To the extent that you are able to find that information, that is.

Mostly I am a a little confused about the interaction between cancer type and tissue type: is every set of markers divided into tissues and which may or may not include cancer types?